src/User/MysqlToggleStatusModifierService.php

```diff
@@ -0,0 +1,317 @@
+            ->setParameter( 0, $param1 )
+            ->setParameter( 1, $param2 );
+        $queryBuilder->execute();
+    }
+
+    /**
+     * @param string $toggleIdentifier
+     * @param string $userOrGroupIdentifier
+     * @param bool $isGroup
+     * @return bool
+     */
+    private function tryInsertElseUpdateToggleToActiveState( $toggleIdentifier, $userOrGroupIdentifier, $isGroup )
+    {
+        if ( !$isGroup ) {
+            $this->tryInsertElseUpdateUserToggleToActiveState( $toggleIdentifier, $userOrGroupIdentifier );
+        } else {
+            $this->tryInsertElseUpdateGroupToggleToAGivenState( $toggleIdentifier, $userOrGroupIdentifier, true );
+        }
+        return true;
+    }
+
+    /**
+     * @param string $toggleIdentifier
+     * @param string $userOrGroupIdentifier
+     * @param bool $isGroup
+     * @return bool
+     */
+    private function tryInsertElseUpdateToggleToInactiveState( $toggleIdentifier, $userOrGroupIdentifier, $isGroup )
+    {
+        if ( !$isGroup ) {
+            $queryBuilder = $this->generateQueryBuilderForUserToggleUpdate();
+            $this->updateToggle( $queryBuilder, $toggleIdentifier, $userOrGroupIdentifier, false );
+        } else {
+            $this->tryInsertElseUpdateGroupToggleToAGivenState( $toggleIdentifier, $userOrGroupIdentifier, false );
+        }
+        return true;
+    }
+
+    /**
+     * @param string $toggleIdentifier
+     * @param string $userOrGroupIdentifier
+     * @param bool $isGroup
+     * @return bool
+     */
+    private function unsetToggle( $toggleIdentifier, $userOrGroupIdentifier, $isGroup )
+    {
+        if ( !$isGroup ) {
+            $queryBuilder = $this->generateQueryBuilderForUserToggleDelete();
+            $this->deleteToggle( $queryBuilder, $toggleIdentifier, $userOrGroupIdentifier );
+        } else {
+            $queryBuilder = $this->generateQueryBuilderForGroupToggleDelete();
+            $this->deleteToggle( $queryBuilder, $toggleIdentifier, $userOrGroupIdentifier );
+        }
+        return true;
+    }
+
+    /**
+     * @param string $toggleIdentifier
+     * @param string $toggleStatus
+     * @param string $userOrGroupIdentifier
+     * @param bool $isGroup
+     * @return bool
+     */
+    private function setToggleStatus( $toggleIdentifier, $toggleStatus, $userOrGroupIdentifier, $isGroup = false )
+    {
+        if ( $this->toggleStatusActive( $toggleStatus ) ) {
+            return $this->tryInsertElseUpdateToggleToActiveState( $toggleIdentifier, $userOrGroupIdentifier, $isGroup );
+        } else if ( $this->toggleStatusInactive( $toggleStatus ) ) {
+            return $this->tryInsertElseUpdateToggleToInactiveState( $toggleIdentifier, $userOrGroupIdentifier,
+                $isGroup );
+        } else if ( $this->toggleStatusUnset( $toggleStatus ) ) {
+            return $this->unsetToggle( $toggleIdentifier, $userOrGroupIdentifier, $isGroup );
+        } else {
+            return false;
+        }
+    }
```

Comment:
just for in the future, you can have the return true after the closing brace of the else: 
``` 
} else { 
    return false;
}
return true;
```
saves having all the returns inside the if statements, but that's not important i dont think. But seeing as they all return true it saves having lots of return points